### DISCUSSION TITLE
How to get time in a variable?

### DISCUSSION THREAD

**Discussion by ChoiZ** [savonet/liquidsoap#2941] [Q&A]
Hello I need help on this little script.

I try to get the current time in a variable to save files and write the current time in a txt file.

My goal is to export meta in a txt file everyday 2023-03-05.txt for today and inside have :
00:00:00 artist - title
00:03:36 artist2 - title2

I have try to do:
```
def write(m)
  file.write(data="#{m[\"on_air\"]} #{m[\"artist\"]} - #{m[\"title\"]}\n", append=true, "%H.txt"))
end
```
For now I have try with %H but it write %H.txt instead of the current hour, and the on_air value is empty.

Thanks for your answers.

**Comment by smimram** [comment]
The function you are looking for is `time.string` which replaces `%H` by the hour and so on:
```
def write(m)
  file.write(data="#{m[\"on_air\"]} #{m[\"artist\"]} - #{m[\"title\"]}\n", append=true, time.string("%H.txt")))
end
```

**Reply by ChoiZ** [reply]
Hello @smimram thank you for your answer.

I have test this but I think it doesn’t works on 1.4.4.

So I have created a var in php and with get_process_output inport into my liq script.

```
def write_meta(m) =
  hour = get_process_output("php hour.php")
  ts = get_process_output("php ts.php")
  print("#{hour}")
  print("#{ts}")
  ignore(file.write(data="#{ts} #{m[\"title\"]}\n", append=true, "#{hour}.txt"))
end
```

On 1.4.4 how to make var for hour and for hour:mn:sec

hour = time() « It’s a timestamp I don’t find in the doc how to deal with it to transform into current hour or h:i:s.

If I can remove the php code It can be better ;-)

Thanks for your help.

**Reply by smimram** [reply]
In 1.4.4, I think that you should be able to work it out with `gettimeofday` and `localtime`: the second take the output of `gettimeofday` and a function which has, as argument, the components of the time. But its not going to be nice, and I'd advise to upgrade your Liquidsoap instead :) I think that something like this should work ot get the hour (and minutes etc can be obtained by changing the `hour` on the right of the line:
```
hour = localtime(gettimeofday(), fun(~sec, ~min, ~hour, ~mday, ~mon, ~year, ~wday, ~yday, ~isdst) -> hour)
```

**Reply by ChoiZ** [reply]
Thank you @smimram it’s working well.

```
hour = localtime(gettimeofday(), fun(~sec, ~min, ~hour, ~mday, ~mon, ~year, ~wday, ~yday, ~isdst) -> hour)
min = localtime(gettimeofday(), fun(~sec, ~min, ~hour, ~mday, ~mon, ~year, ~wday, ~yday, ~isdst) -> min)
sec = localtime(gettimeofday(), fun(~sec, ~min, ~hour, ~mday, ~mon, ~year, ~wday, ~yday, ~isdst) -> sec)
```

For me it’s quite complicated this notation with fun(). Thank you for your help, yes I have migrated some script to the last version but not this one yet.

Is it possible to format min & sec to have initial 0 ?
